plugins/hls-refactor-plugin/src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -520,7 +520,7 @@ caRemoveInvalidExports m contents digs ctxDigs uri
         _documentChanges = Nothing
         _edit = Just WorkspaceEdit{..}
         _command = Nothing
-        _isPreferred = Nothing
+        _isPreferred = Just True
```

Comment:
This is a policy decision. You made a good argument for why we should do this in the PR description, but all that information is going to be lost once this is merged. Write it down! I think this is a good place for a Note-style comment, e.g.

```
-- Note [Removing imports is preferred]
-- It's good to prefer the remove imports code action because...
...
   -- See Note [Removing imports is preferred]
   _isPreferred = Just True

Reply:
> You made a good argument for why we should do this in the PR description, but all that information is going to be lost once this is merged

`git blame` on the line will show the explanation as part of the commit and refer to the pull request. I was thinking about a comment, but assumed we'll have many code actions that need to be adapted like this, and I thought comments would be two much. But if that's the preferred granularity and place for documentation, I'll add one here.


https://user-images.githubusercontent.com/13085980/190390388-72bfa8ee-97a5-4c96-a592-a9bc80c9ad25.mp4

Reply:
Commit messages last until someone else touches that line. Plus, maybe someone wants to change it *and* change the reasoning. IMO there is no substitute for just writing things in the code!

Reply:
More of my opinions if you care ;) https://www.michaelpj.com/blog/2022/04/24/on-commenting-code.html

Reply:
Interesting, I can see arguments for both ways.

> Commit messages last until someone else touches that line.

"Put it in code" often comes from times where tooling was bad or didn't exist. Git blame / commits can be viewed before the change as well, and there is history and references to issue and PR. But this only works well for small incremental changes (and not large commits changing everything). So it depends on the project.

![image](https://user-images.githubusercontent.com/13085980/190418763-ce976a7d-5161-4992-aaf5-6a4ec07fdc4e.png)

What I've seen more often is that code will be changed, but the comment will not be adapted and be misleading/wrong. In these cases newer commits overlaying older ones isn't necessarily bad.

The most important part is *that there is* is place where a decision is documented at all and it's somewhat consistently done in the project (format and granularity). Not really arguing to do anything differently here, just bringing up a different view 😀